### ISSUE TITLE
Code folding based on indentation

### ISSUE THREAD

**Issue by goulashsoup** [massivemadness/Squircle-CE#41]
Something that is missing in all android code editors is the possibility to fold based on indendation (and selection).

I'd like to work on that feature but before doing so i'd like to know if such a feature would be accepted?

**Comment by massivemadness** [comment]
Yeah, it would be awesome to have it like in IDEA (simply `+` and `-` buttons). But I don't know how you going to do that, because that's something that should be pretty optimized in some way 🤷‍♂️
